daemon/openrazer_daemon/daemon.py

```diff
@@ -479,10 +491,25 @@ def _udev_input_event(self, device):
         """
         self.logger.debug('Device event [%s]: %s', device.action, device.device_path)
         if device.action == 'add':
-            self._add_device(device)
+            if self._collecting_udev:
+                self._collecting_udev_devices.append(device)
+                return
+            else:
+                self._collecting_udev_devices = [device]
+                self._collecting_udev = True
+                t = threading.Thread(target=self._collecting_udev_method, args=(device,))
+                t.start()
         elif device.action == 'remove':
             self._remove_device(device)
 
+    def _collecting_udev_method(self, device):
+        time.sleep(2)  # delay to let udev add all devices that we want
+        # Sort the devices
+        self._collecting_udev_devices.sort(key=lambda x: x.sys_path, reverse=True)
+        for d in self._collecting_udev_devices:
+            self._add_device(d)
+        self._collecting_udev = False
```

Comment:
IMO you're just moving the problem here to occur less often than before. The device should be capable of having interfaces removed and added at *any* time and do the right thing. That way you don't have to care about the delay or any other timeouts.

Reply:
That would be the optimal solution but in my limited testing on one computer it worked always (the delay is actually quite a bit longer than neccessary).
Otherwise for the daemon to handle it would require quite a bit of restructuring of the code there and I'm not really keen on doing that ^^